ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/tasks/PipelinedImportChainSegmentTask.java

```diff
@@ -38,7 +40,8 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class PipelinedImportChainSegmentTask<C, B> extends AbstractEthTask<List<B>> {
+public class PipelinedImportChainSegmentTask<C, B extends BlockParts>
```

Comment:
BlockResults will be used by a future task.  I can move it back into the sync package but keeping it in the PipelinedImportChainSegmentTask won't work in the future, it's not specific to that class.

Reply:
Also, it's used in the `fastsync` and `fullsync` packages, but the fully depend on the sync package.  File moved.

Reply:
Sorry - I think my last comment was a bit vague and probably posted in a confusing spot.  Looks like you took this as a comment on the location of the `BlockHandler` class?  I meant to suggest renaming `BlockParts` to `BlockResult` and moving `BlockResult` into `PipelinedImportChainSegment` or at least into the `eth.sync` (or `eth.sync.tasks`) package.

Reply:
That quickly begs the question of why have a generic, and I would need to undo alot of Adrian's changes to make it work.  I can prototype it out but I don't know how not having a generic will impact Adrian's work.

Reply:
I think I'm done in that area of code os don't mind if it massively changes as long as it keeps working now. :)

Reply:
Changing the generics is a large undertaking, 8+ files and I'm still not done chasing the tendrils.

Moving `BlockParts`/`BlockResult` into the `PipelineImportChainSegment` class is a bad design idea.  `Block` implements it and then would depend on the `eth.sync` class, and we would tie all the data objects into depending on the sync logic, which from a design standpoint is bad.  I think that `BlockParts`/`BlockResult` is in the correct place.  It makes the functionality accessible without making the `Block` class depend on the import logic.

Reply:
I'm definitely not suggesting making `Block` depend on the `eth.sync` package.  I just don't think that we should be adding core interfaces for sync-specific changes that are isolated to the sync package.  And `BlockParts` is really another way of saying `Block` - it's not a new or fundamental concept.

I think you should just be able to have the generic param `B` bound by a new class (say `BlockResult`) that exposes whatever common methods you need.  You can then modify the `BlockHandler.downloadBlocks()` methods to wrap your result in a `BlockResult` or return a new type or whatever.  This doesn't seem like a huge change to me - but maybe I'm missing something?

Reply:
Rather than a separate wrapper, this most recent push imply uses BlockWithReceipts and moves it as far out as needed so we aren't wrapping/unwrapping in completable future stage transitions.  This also eliminates the need for the generic object and the need to put bounds on it.